### PR TITLE
Customer Assigned Fault Domain on VM referencing VMSS - SDK and Tests

### PR DESCRIPTION
**Pull request by shpa-microsoft**
Api pull request https://github.com/Azure/azure-rest-api-specs/pull/11409

Do not merge, this PR is to confirm that .Net SDK code can be generated from Swagger change.

### CODE REVIEW

**Review by markcowl — CHANGES_REQUESTED** [review]
The code changes look OK, but there are some things that are missing:
- Please add in a reference to the corresponding Swagger PR for this change
- running the generation script produces an updated metadata file in the eng directory, please add this in to the PR
- When api versions change, you need to update Az.Sdk.RP.props to change the tag to match the new api-version
- You sill need to re-record the tests and add in new tests for any new functionality

**Comment by markcowl** [comment]
@shpa-microsoft Also, please ensure you update this branch by pulling in the latest from master.

**Comment by shpa-microsoft** [comment]
@markcowl Addressed your comments with some questions on offline chat.

**Comment by shpa-microsoft** [comment]
Can't perform tests due to ARM API release issues. Spoke offline about it.

**Comment by jsquire** [comment]
Hi @shpa-microsoft.  There hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream or if the PR should be closed out?

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs [hunk 1]
```diff
@@ -134,7 +138,7 @@ public async Task<AzureOperationResponse<Gallery>> UpdateWithHttpMessagesAsync(s
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<Gallery>> GetWithHttpMessagesAsync(string resourceGroupName, string galleryName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<Gallery>> GetWithHttpMessagesAsync(string resourceGroupName, string galleryName, string select = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs:141]: Need a customization to fix this breaking change - essentially you need a method that implements the old parameters, with 'customHeaders' being non-optional, like the following:

```
public partial class GalleriesOperations {

    public async Task<AzureOperationResponse<Gallery>> GetWithHttpMessagesAsync(string resourceGroupName, string galleryName, Dictionary<string, List<string>> customHeaders, CancellationToken cancellationToken = default(CancellationToken))
    {
         return GetWithHttpMessagesAsync(resourceGroupName, galleryName, null, customHeaders, cancellationToken);
    }
}
```

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs [hunk 2]
```diff
@@ -150,12 +154,16 @@ public static Gallery Get(this IGalleriesOperations operations, string resourceG
...
             /// </param>
+            /// <param name='select'>
+            /// The select expression to apply on the operation. Possible values include:
+            /// 'Permissions'
+            /// </param>
             /// <param name='cancellationToken'>
             /// The cancellation token.
             /// </param>
-            public static async Task<Gallery> GetAsync(this IGalleriesOperations operations, string resourceGroupName, string galleryName, CancellationToken cancellationToken = default(CancellationToken))
+            public static async Task<Gallery> GetAsync(this IGalleriesOperations operations, string resourceGroupName, string galleryName, string select = default(string), CancellationToken cancellationToken = default(CancellationToken))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs:164]: same comment as above

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachineScaleSetVMExtensionsOperations.cs [hunk 3]
```diff
@@ -57,7 +57,7 @@ public partial interface IVirtualMachineScaleSetVMExtensionsOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<VirtualMachineExtension>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, string vmExtensionName, VirtualMachineExtension extensionParameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachineScaleSetVMExtensionsOperations.cs:60]: need to compensate for these breaking changes:

(1) ensure that the return type has all the properties of the previous return type
(2) add method overloads through customizing thre partial interface using the old signature

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachinesOperations.cs [hunk 4]
```diff
@@ -154,7 +157,7 @@ public partial interface IVirtualMachinesOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string vmName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string vmName, bool? forceDeletion = default(bool?), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachinesOperations.cs:160]: Need to add overloads as above to fix these breakign changes

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryApplicationVersionPublishingProfile.cs [hunk 5]
```diff
@@ -78,11 +75,9 @@ public GalleryApplicationVersionPublishingProfile(UserArtifactSource source, ILi
         public UserArtifactSource Source { get; set; }
 
         /// <summary>
-        /// Gets or sets optional. May be used to help process this file. The
-        /// type of file contained in the source, e.g. zip, json, etc.
         /// </summary>
-        [JsonProperty(PropertyName = "contentType")]
-        public string ContentType { get; set; }
+        [JsonProperty(PropertyName = "manageActions")]
+        public UserArtifactManage ManageActions { get; set; }
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryApplicationVersionPublishingProfile.cs:80]: This is a breaking change - you will have to maintain a ContentType property, even if unused

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImage.cs [hunk 6]
```diff
@@ -49,22 +49,23 @@ public GalleryImage()
...
+        /// image definition.</param>
+        public GalleryImage(string location, OperatingSystemTypes osType, OperatingSystemStateTypes osState, GalleryImageIdentifier identifier, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), string description = default(string), string eula = default(string), string privacyStatementUri = default(string), string releaseNoteUri = default(string), string hyperVGeneration = default(string), IList<GalleryImageFeature> features = default(IList<GalleryImageFeature>), System.DateTime? endOfLifeDate = default(System.DateTime?), RecommendedMachineConfiguration recommended = default(RecommendedMachineConfiguration), Disallowed disallowed = default(Disallowed), ImagePurchasePlan purchasePlan = default(ImagePurchasePlan), string provisioningState = default(string))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImage.cs:68]: Need to add a method overload to fix the breaking change, as above

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImageUpdate.cs [hunk 7]
```diff
@@ -48,22 +48,23 @@ public GalleryImageUpdate()
...
+        /// image definition.</param>
+        public GalleryImageUpdate(OperatingSystemTypes osType, OperatingSystemStateTypes osState, GalleryImageIdentifier identifier, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), string description = default(string), string eula = default(string), string privacyStatementUri = default(string), string releaseNoteUri = default(string), string hyperVGeneration = default(string), IList<GalleryImageFeature> features = default(IList<GalleryImageFeature>), System.DateTime? endOfLifeDate = default(System.DateTime?), RecommendedMachineConfiguration recommended = default(RecommendedMachineConfiguration), Disallowed disallowed = default(Disallowed), ImagePurchasePlan purchasePlan = default(ImagePurchasePlan), string provisioningState = default(string))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImageUpdate.cs:67]: Need to add an additional overload through customizations to fix this issue

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/UserArtifactSource.cs [hunk 8]
```diff
@@ -46,19 +47,20 @@ public UserArtifactSource(string fileName, string mediaLink)
         /// </summary>
         partial void CustomInit();
 
-        /// <summary>
-        /// Gets or sets required. The fileName of the artifact.
-        /// </summary>
-        [JsonProperty(PropertyName = "fileName")]
-        public string FileName { get; set; }
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/UserArtifactSource.cs:53]: Cannot trmoeve this property - must maintain the property

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMExtensionsOperations.cs [hunk 9]
```diff
@@ -74,10 +74,10 @@ internal VirtualMachineScaleSetVMExtensionsOperations(ComputeManagementClient cl
         /// <param name='cancellationToken'>
         /// The cancellation token.
         /// </param>
-        public async Task<AzureOperationResponse<VirtualMachineExtension>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, string vmExtensionName, VirtualMachineExtension extensionParameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<VirtualMachineScaleSetVMExtension>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, string vmExtensionName, VirtualMachineScaleSetVMExtension extensionParameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMExtensionsOperations.cs:77]: Same issues as above:
(1) Need to verify that return tyoe contains all the properties of the previous return type
(2) Need to add a customization to implement the old parameter set, as above.

This applies for all the similar breaking method signature changes in this file

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMExtensionsOperationsExtensions.cs [hunk 10]
```diff
@@ -42,7 +42,7 @@ public static partial class VirtualMachineScaleSetVMExtensionsOperationsExtensio
             /// <param name='extensionParameters'>
             /// Parameters supplied to the Create Virtual Machine Extension operation.
             /// </param>
-            public static VirtualMachineExtension CreateOrUpdate(this IVirtualMachineScaleSetVMExtensionsOperations operations, string resourceGroupName, string vmScaleSetName, string instanceId, string vmExtensionName, VirtualMachineExtension extensionParameters)
+            public static VirtualMachineScaleSetVMExtension CreateOrUpdate(this IVirtualMachineScaleSetVMExtensionsOperations operations, string resourceGroupName, string vmScaleSetName, string instanceId, string vmExtensionName, VirtualMachineScaleSetVMExtension extensionParameters)
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMExtensionsOperationsExtensions.cs:45]: Same as in previous file - need to add customizations to fix these breaking changes

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachinesOperationsExtensions.cs [hunk 11]
```diff
@@ -230,12 +233,15 @@ public static void Delete(this IVirtualMachinesOperations operations, string res
...
             /// The name of the virtual machine.
             /// </param>
+            /// <param name='forceDeletion'>
+            /// Optional parameter to force delete virtual machines.
+            /// </param>
             /// <param name='cancellationToken'>
             /// The cancellation token.
             /// </param>
-            public static async Task DeleteAsync(this IVirtualMachinesOperations operations, string resourceGroupName, string vmName, CancellationToken cancellationToken = default(CancellationToken))
+            public static async Task DeleteAsync(this IVirtualMachinesOperations operations, string resourceGroupName, string vmName, bool? forceDeletion = default(bool?), CancellationToken cancellationToken = default(CancellationToken))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachinesOperationsExtensions.cs:242]: need to add an overload to fix this breaking change

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachinesOperationsExtensions.cs [hunk 12]
```diff
@@ -1204,12 +1213,15 @@ public static void BeginDelete(this IVirtualMachinesOperations operations, strin
...
             /// The name of the virtual machine.
             /// </param>
+            /// <param name='forceDeletion'>
+            /// Optional parameter to force delete virtual machines.
+            /// </param>
             /// <param name='cancellationToken'>
             /// The cancellation token.
             /// </param>
-            public static async Task BeginDeleteAsync(this IVirtualMachinesOperations operations, string resourceGroupName, string vmName, CancellationToken cancellationToken = default(CancellationToken))
+            public static async Task BeginDeleteAsync(this IVirtualMachinesOperations operations, string resourceGroupName, string vmName, bool? forceDeletion = default(bool?), CancellationToken cancellationToken = default(CancellationToken))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachinesOperationsExtensions.cs:1222]: need to add an overload to fix this breaking change

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by markcowl** [comment]
@shpa-microsoft Marked the breaking changes that need to be fixed through customization

**Comment by markcowl** [comment]
@shpa-microsoft It looks like this PR is out of date, can you fix the merge conflicts?

**Comment by shpa-microsoft** [comment]
@markcowl all the comments made for breaking changes are unrelated to my changes; and generated automatically with autorest. Is there a way I only regenerate my code for this PR?

**Comment by markcowl** [comment]
@shpa-microsoft  There is a discussion about these API changes, and a proposed resolution on the compute team.  Please contact @Sandido for details.

**Pull request closed without merging**